assets/src/blocks/Gallery/GalleryCarousel.js

```diff
@@ -1,59 +1,124 @@
-import { Component } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 import _uniqueId from 'lodash/uniqueId';
 
 const { __ } = wp.i18n;
 
-export class GalleryCarousel extends Component {
-  constructor(props) {
-    super(props);
-    this.id = `gallery_${_uniqueId()}`;
+const reflow = element => element.offsetHeight;
+
+export const GalleryCarousel = ({ images }) => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [sliding, setSliding] = useState(false);
+
+  const lastSlide = images.length - 1;
+
+  const getOrder = newSlide => {
+    let order = newSlide < currentSlide ? 'prev' : 'next';
+    if (newSlide === lastSlide && currentSlide === 0 && order !== 'prev') {
+      order = 'prev';
+    } else if (newSlide === 0 && currentSlide === lastSlide && order !== 'next') {
+      order = 'next';
+    }
+    return order;
+  }
+
+  const goToSlide = newSlide => {
+    if (newSlide !== currentSlide && !sliding) {
+      setSliding(true);
+      const order = getOrder(newSlide);
+      const direction = order === 'next' ? 'left' : 'right';
+      const nextElement = document.getElementById(`${idRef.current}-slide${newSlide}`);
+      const activeElement = document.getElementById(`${idRef.current}-slide${currentSlide}`);
+      const orderClassname = `carousel-item-${order}`;
+      const directionClassname = `carousel-item-${direction}`;
+
+      nextElement.classList.add(orderClassname);
```

Comment:
I have a `nextElement does not exist` or something like that when I insert a brand new gallery, before hitting "Insert gallery" but after selecting the images (it probably creates the gallery view before having a list of images ?), which stops interactivity of the gallery; fixed it with this diff, don't know if there's a better solution :man_shrugging: 
```diff
diff --git a/assets/src/blocks/Gallery/GalleryCarousel.js b/assets/src/blocks/Gallery/GalleryCarousel.js
index a55dc7c..b7ed24f 100644
--- a/assets/src/blocks/Gallery/GalleryCarousel.js
+++ b/assets/src/blocks/Gallery/GalleryCarousel.js
@@ -31,16 +31,16 @@ export const GalleryCarousel = ({ images }) => {
       const orderClassname = `carousel-item-${order}`;
       const directionClassname = `carousel-item-${direction}`;
 
-      nextElement.classList.add(orderClassname);
+      nextElement?.classList.add(orderClassname);
 
-      reflow(nextElement);
+      nextElement ? reflow(nextElement) : null;
 
-      activeElement.classList.add(directionClassname);
-      nextElement.classList.add(directionClassname);
+      activeElement?.classList.add(directionClassname);
+      nextElement?.classList.add(directionClassname);
 
       setTimeout(() => {
-        nextElement.classList.remove(directionClassname, orderClassname);
-        activeElement.classList.remove(orderClassname, directionClassname);
+        nextElement?.classList.remove(directionClassname, orderClassname);
+        activeElement?.classList.remove(orderClassname, directionClassname);
         setSliding(false);
         setCurrentSlide(newSlide);
       }, 600);

```

Reply:
Good catch! I've pushed a fix for it, a bit different from yours, that I think is a bit easier to read 😉 I've added a check in the `useEffect` for autoplay to apply it only if `images.length` is more than 1. And to be extra safe I also check in `goToSlide` that `nextElement` and `activeElement` are not null!